add_me_to_your_PATH/Dockerfile.template

```diff
@@ -5,15 +5,24 @@ ARG CUDA_VERSION
 
 FROM julia:${JULIA_VERSION}-buster as pinned-extractor
 
-COPY *Project.toml *Manifest.toml .
+COPY Project.toml Manifest.toml Manifest.toml.julia_pod .
 
 # Remove all packages that are not pinned recursively from the manifest
 # (stdlibs count as pinned)
 RUN julia --project=. -e 'using Pkg;\
-                          unpinned = [p.name for p in values(Pkg.dependencies()) \
-                                      if p.is_direct_dep && !p.is_pinned && \
-                                      p.name ∉ values(Pkg.Types.stdlibs())]; \
-                          isempty(unpinned) || Pkg.rm(unpinned; mode=Pkg.PKGMODE_MANIFEST)'
+                          unregistered = [p.name for p in values(Pkg.dependencies()) \
+                                          if !p.is_tracking_registry]; \
+                          registered = [p.name for p in values(Pkg.dependencies()) \
+                                        if p.is_tracking_registry]; \
+                          previous = [p.name for p in values(Pkg.Types.read_manifest("Manifest.toml.julia_pod"))]; \
+                          rm_deps = union(unregistered, setdiff(registered, previous)); \
+                          println("removing $rm_deps"); sleep(1); \
+                          isempty(rm_deps) || Pkg.rm(rm_deps; mode=Pkg.PKGMODE_MANIFEST); \
+                          direct = [p.name for p in values(Pkg.dependencies()) \
+                                          if !p.is_direct_dep]; \
+                          intersect!(rm_deps, direct); \
+                          isempty(rm_deps) || Pkg.rm(rm_deps; mode=Pkg.PKGMODE_PROJECT); \
+                          println(filter(contains(Regex(join(rm_deps, "|"))), readlines("Project.toml")))'
```

Comment:
Yup, `]rm` on a dep that was previously in the sysimage invalidates the cache, because that makes for a different sysimage-private `Project.toml`. I've added a sentence to the `README.md` about this.